sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs

```diff
@@ -29,10 +37,8 @@ public Uri AuthorityHost
         public bool AllowMultiTenantAuthentication { get; set; }
 
         /// <summary>
-        /// Gets or sets value indicating if ETW logging that contains PII content should be logged.
-        /// Setting this property will not disable redaction of <see cref="Request"/> Content. To enable logging of sensitive <see cref="Request.Content"/>
-        /// the <see cref="DiagnosticsOptions.IsLoggingContentEnabled"/> property must be set to <c>true</c>.
+        /// Gets the credential diagnostic options.
         /// </summary>
-        public bool IsLoggingPIIEnabled { get; set; }
+        public new TokenCredentialDiagnosticsOptions Diagnostics => base.Diagnostics as TokenCredentialDiagnosticsOptions;
```

Comment:
I don't think we should use new. It has bad semantics. See
https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUBRQG4TIAEA9gA4IDO9AvPTAKYDu9ACqMA1rxgBhADZQJCAPKsojGGwAUASgDcFFuwB0AESgQA5jEZsEUAMZKbqtgYCSbADKMzZ2GYCiMBDA0rwYXPQIaLy6lOQycjCKyk70tvoc3OkxFPjEAJzqaazOJuaW1nYOKmquHl4+MP6BwaE6VORE9PHyVU4UAN4U9MP0+ADM9KUWVjb2yWqTptMVc44L/Wa8CNoAvuF8/FoxO+2dU+WzvWoDQyPj9MCMjNL0bp7evgFBIWEbW9psf4ncjAnKEYRiCTdRJXNggLqyHrzNg3cgjUYTA4Q8Qwc4zSrIxZlfGraocP7bPbcA5HCigjrgkQ4vErWHwlmXZGo9H3R7PV5sABiEBgtgAnu8Gk1vqF6BSAUC6RQgA

Reply:
Or we at least need to make sure that we are propagating any changes to the base class options.

Reply:
That is  why I made the set protected and actually set the object on the base class. I agree with you that new is not ideal, but I think it's safer in this case since the accessor is just casting.

Reply:
Missed that